r-package/grf/R/local_linear_forest.R

```diff
@@ -0,0 +1,155 @@
+                              num.fit.trees = 10,
+                              num.fit.reps = 100,
+                              num.optimize.reps = 1000) {
+
+  # return a standard regression forest with appropriate defaults
+  forest = regression_forest(X, Y, sample.fraction,
+                             mtry,
+                             num.trees,
+                             num.threads,
+                             min.node.size,
+                             honesty,
+                             ci.group.size = 1,
+                             alpha,
+                             imbalance.penalty,
+                             compute.oob.predictions,
+                             seed,
+                             clusters,
+                             samples_per_cluster,
+                             tune.parameters,
+                             num.fit.trees,
+                             num.fit.reps,
+                             num.optimize.reps)
+
+  forest
+}
+
+#' Predict with a local linear forest
+#'
+#' Gets estimates of E[Y|X=x] using a trained regression forest.
+#'
+#' @param object The trained forest.
+#' @param newdata Points at which predictions should be made. If NULL,
+#'                makes out-of-bag predictions on the training set instead
+#'                (i.e., provides predictions at Xi using only trees that did
+#'                not use the i-th training example).
+#' @param linear.correction.variables Optional subset of indexes for variables to be used in local
+#'                   linear prediction. If left NULL, all variables are used.
+#'                   We run a locally weighted linear regression on the included variables.
+#'                   Please note that this is a beta feature still in development, and may slow down
+#'                   prediction considerably. Defaults to NULL.
+#' @param ll.lambda Ridge penalty for local linear predictions
+#' @param tune.lambda Optional self-tuning for ridge penalty lambda. Defaults to FALSE.
+#' @param lambda.path Optional list of lambdas to use for cross-validation, used if tune.lambda is TRUE.
+#' @param ll.ridge.type Option to standardize ridge penalty by covariance ("standardized"),
+#'                   or penalize all covariates equally ("identity").
+#' @param num.threads Number of threads used in training. If set to NULL, the software
+#'                    automatically selects an appropriate amount.
+#' @param ... Additional arguments (currently ignored).
+#'
+#' @return A vector of predictions.
+#'
+#' @examples \dontrun{
+#' # Train the forest.
+#' n = 50; p = 5
+#' X = matrix(rnorm(n*p), n, p)
+#' Y = X[,1] * rnorm(n)
+#' forest = local_linear_forest(X, Y)
+#'
+#' # Predict using the forest.
+#' X.test = matrix(0, 101, p)
+#' X.test[,1] = seq(-2, 2, length.out = 101)
+#' predictions = predict(forest, X.test)
+#'
+#' # Predict on out-of-bag training samples.
+#' predictions.oob = predict(forest)
+#' }
+#'
+#' @export
```

Comment:
Does this function ever get called? One would need to make sure relevant forests are of class `local_linear_forest` first; see https://adv-r.hadley.nz/s3.html#classes for more.